applications/grpc_hello_world/grpc_client.cpp

```diff
@@ -0,0 +1,138 @@
+ public:
+  HOLOSCAN_OPERATOR_FORWARD_ARGS(PrintOperator)
+
+  PrintOperator() = default;
+  void setup(OperatorSpec& spec) override { spec.input<std::string>("greeting"); }
+  void compute(InputContext& op_input, OutputContext&, ExecutionContext& context) override {
+    auto reply = op_input.receive<std::string>("greeting");
+    if (reply) { HOLOSCAN_LOG_INFO("Greeter received: {}", reply.value()); }
+  }
+};
+
+class SayMyNameOperator : public holoscan::Operator {
+ public:
+  HOLOSCAN_OPERATOR_FORWARD_ARGS(SayMyNameOperator)
+
+  SayMyNameOperator() = default;
+
+  void start() override {}
+
+  void setup(OperatorSpec& spec) override { spec.output<std::string>("name"); }
+
+  void compute(InputContext& op_input, OutputContext& op_output,
+               ExecutionContext& context) override {
+    std::string user("world");
+    op_output.emit(user, "name");
+  }
+};
+
+template <typename GrpClientT>
+class GrpcClientOperator : public holoscan::Operator {
+ public:
+  HOLOSCAN_OPERATOR_FORWARD_ARGS(GrpcClientOperator);
+
+  GrpcClientOperator() = default;
+
+  void start() override {
+    HOLOSCAN_LOG_INFO("Starting gRPC client...");
+    grpc_client_ = std::make_shared<GrpClientT>(
+        GrpClientT(grpc::CreateChannel(server_address_.get(), grpc::InsecureChannelCredentials())));
+  }
+
+  void setup(OperatorSpec& spec) override {
+    spec.param<std::string>(
+        server_address_, "server_address", "gRPC server address", "gRPC server address");
+
+    spec.input<std::string>("name");
+    spec.output<std::string>("greeting");
+  }
+
+  void compute(InputContext& op_input, OutputContext& op_output,
+               ExecutionContext& context) override {
+    auto name = op_input.receive<std::string>("name");
+
+    if (name) {
+      std::string reply = grpc_client_->HelloWorld(name.value());
+      op_output.emit(reply, "greeting");
+    }
+  }
+
+ private:
+  Parameter<std::string> server_address_;
+  std::shared_ptr<GrpClientT> grpc_client_;
+};
+
+//
+
+class ClientApp : public holoscan::Application {
+ public:
+  explicit ClientApp(const std::string& server_address) : server_address_(server_address) {}
+  void compose() override {
+    auto say_my_name =
+        make_operator<SayMyNameOperator>("say_my_name", make_condition<CountCondition>(1));
+    auto grpc_client = make_operator<GrpcClientOperator<EntityClient>>(
+        "grpc_client", Arg("server_address", server_address_));
+    auto print_name = make_operator<PrintOperator>("print_name");
+
+    add_flow(say_my_name, grpc_client);
+    add_flow(grpc_client, print_name);
+  }
+
+ private:
+  std::string server_address_;
+};
+}  // namespace grpc_hello_world
+}  // namespace holoscan
+
+int main(int argc, char** argv) {
+  auto config_path = std::filesystem::canonical(argv[0]).parent_path();
+  config_path += "/grpc_hello_world.yaml";
+  auto app = holoscan::make_application<holoscan::grpc_hello_world::ClientApp>("localhost:50051");
+  app->config(config_path);
```

Comment:
Thoughts on accepting a custom config path to aid testing? Not needed for this app scope?